### PR TITLE
Feat(eos_designs): Add support for subinterfaces in nodes.l3_interfaces

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

Implement same auto dectection of subinterfaces as in `network_services.l3_interfaces` in `node.l3_interfaces`

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

Just as for network_services, if a subinterface is configured, the parent physical interface will be automatically configured as well.

## How to test

molecule

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-transit.cfg [hunk 1]
```diff
@@ -223,13 +223,21 @@ interface Dps1
 interface Ethernet1
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-transit.cfg:225]: I think you may need to handle MTU somehow, but we can see later. (add the max mtu of subinterfaces on the generated parent)

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-transit.cfg:225]: We don't support MTU on these interfaces currently, so I will resolve this for now.

**Review by ClausHolbechArista — APPROVED** [review]

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/utils.py [hunk 2]
```diff
@@ -137,6 +137,11 @@ def _get_l3_interface_cfg(self, l3_interface: dict) -> dict | None:
         Returns structured_configuration for one L3 interface
         """
         interface_name = get(l3_interface, "name", required=True, org_key=f"<node_type_key>...[node={self.shared_utils.hostname}].l3_interfaces[].name]")
+        if "." in interface_name:
+            iface_type = "l3dot1q"
+            encapsulation = get(l3_interface, "encapsulation_dot1q_vlan", default=interface_name.split(".")[-1])
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/utils.py]: Encapsulation should be an integer.
```suggestion
            encapsulation = int(get(l3_interface, "encapsulation_dot1q_vlan", default=interface_name.split(".")[-1]))
```

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/structured_configs/cv-pathfinder-transit.yml [hunk 3]
```diff
@@ -136,21 +136,31 @@ management_api_http:
...
   enable_https: true
 ethernet_interfaces:
-- name: Ethernet1
+- name: Ethernet1.42
   peer_type: l3_interface
   ip_address: dhcp
   shutdown: false
-  type: routed
+  type: l3dot1q
+  encapsulation_dot1q_vlan: '42'
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/structured_configs/cv-pathfinder-transit.yml]: Encapsulation should be an interger.
```suggestion
  encapsulation_dot1q_vlan: 42
```

**Review by carlbuchmann — APPROVED** [review]
LGTM, after correcting the type when auto generated. Tested in lab topology.

**Pull request merged**